src/Framework/Sdk/SdkResultItem.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// The value of an item and any associated metadata to be added by an SDK resolver.  See <see cref="SdkResult.ItemsToAdd"/>
+    /// </summary>
+    public class SdkResultItem
+    {
+        public string ItemSpec { get; set; }
+        public Dictionary<string, string> Metadata { get;}
+
+        public SdkResultItem()
+        {
+            Metadata = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Creates an <see cref="SdkResultItem"/>
+        /// </summary>
+        /// <param name="itemSpec">The value (itemspec) for the item</param>
+        /// <param name="metadata">A dictionary of item metadata.  This should be created with <see cref="StringComparer.OrdinalIgnoreCase"/> for the comparer.</param>
```

Comment:
Not `MSBuildNameIgnoreCaseComparer`?

Reply:
Debug-only assert for this invariant?

Reply:
I wasn't aware of that comparer, but it's internal and not in the Microsoft.Build.Framework assembly, so we can't tell people to use it.